pydal/objects.py

```diff
@@ -2172,14 +2172,15 @@ def build(self, d):
                              d.get("second", None))
         left = right = built = None
 
-        if op in ("AND", "OR"):
+        op = op.upper()
```

Comment:
I understand this fix. But why change AND -> _AND?

Reply:
When the query is pickle to the session the operators *and*, *or* and *not* are prefixed with `_`, so: `_AND`, `_OR` and `_NOT`

```python
In [1]: q = db(db.auth_user.id > 0) & (db.auth_user.id < 10)

In [2]: q
Out[2]: <Query (("auth_user"."id" < 10) AND <Set ("auth_user"."id" > 0)>)>

In [3]: q.as_dict()
Out[3]: 
{'_db': <DAL uri="sqlite://storage.sqlite">,
 'db': <DAL uri="sqlite://storage.sqlite">,
 'first': <Query ("auth_user"."id" < 10)>,
 'ignore_common_filters': False,
 'op': <bound method SQLiteDialect._and of <pydal.dialects.sqlite.SQLiteDialect object at 0x7f897554ff90>>,
 'optional_args': {},
 'second': <Set ("auth_user"."id" > 0)>}

In [4]: q.as_dict(flat=True)
Out[4]: 
{'first': {'first': {'fieldname': 'id', 'tablename': 'auth_user'},
  'ignore_common_filters': False,
  'op': 'lt',
  'optional_args': {},
  'second': 10},
 'ignore_common_filters': False,
 'op': '_and',
 'optional_args': {}}
```

Note to that the `<` is not prefixed with `_`. Actually all the operatos are lowercase, i just make then upper there for the sake of not making to many changes.

Reply:
Sorry I am slow I still do not understand where do the '_' comes from?